jina/peapods/remote.py

```diff
@@ -47,21 +47,20 @@ def spawn_remote(self, host: str, port: int, **kwargs) -> Optional[str]:
 
     def loop_body(self):
         if self.remote_id:
-            self.logger.success(f'created remote {self.api.kind} with id {colored(self.remote_id, "cyan")}')
-            self.set_ready()
-            self.api.log(self.remote_id)
+            try:
+                self.logger.success(f'created remote {self.api.kind} with id {colored(self.remote_id, "cyan")}')
+                self.set_ready()
+                self.api.log(self.remote_id)
+            finally:
+                self.is_shutdown.set()
         else:
             self.logger.error(f'fail to create {typename(self)} remotely')
             self.is_shutdown.set()
 
     def delete_remote(self):
         if hasattr(self, 'api') and self.api.is_alive and self.remote_id:
             self.api.delete(self.remote_id)
-
-    def close(self):
-        self.delete_remote()
-        self.join()
```

Comment:
I think this join would never work right? or it would work when the log will end?